python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -296,22 +300,25 @@ def asset_key_to_dep_node_handles(
         assets_defs_by_node_handle=assets_defs_by_node_handle,
     )
 
-    dep_nodes_by_asset_key: Dict[AssetKey, List[NodeHandle]] = {}
-    dep_node_outputs_by_asset_key: Dict[AssetKey, List[NodeOutputHandle]] = {}
+    dep_nodes_by_asset_or_check_key: Dict["AssetKeyOrCheckKey", List[NodeHandle]] = {}
+    dep_node_outputs_by_asset_or_check_key: Dict["AssetKeyOrCheckKey", List[NodeOutputHandle]] = {}
 
     for node_handle, assets_defs in assets_defs_by_node_handle.items():
         dep_node_output_handles_by_node: Dict[
             NodeOutputHandle, Sequence[NodeOutputHandle]
         ] = {}  # memoized map of node output handles to all node output handle dependencies that are from ops
-        for output_name, asset_key in assets_defs.keys_by_output_name.items():
-            dep_nodes_by_asset_key[
-                asset_key
+        for (
+            output_name,
+            asset_or_check_key,
+        ) in assets_defs.asset_and_check_keys_by_output_name.items():
+            dep_nodes_by_asset_or_check_key[
+                asset_or_check_key
             ] = []  # first element in list is node that outputs asset
 
-            dep_node_outputs_by_asset_key[asset_key] = []
+            dep_node_outputs_by_asset_or_check_key[asset_or_check_key] = []
```

Comment:
Looks like none of the callsites actually require `dep_node_outputs_by_asset_or_check_key` to contain check information, can we instead exclusively change ` dep_nodes_by_asset_or_check_key`?

(I think we only need to add check information to `dep_node_outputs_by_asset_or_check_key` if we needed to support calling `context.selected_output_names` from within a check op)

Reply:
Without it we hit `Could not find output mapping asset_one_check1` when resolving the job